analyzers/tests/SonarAnalyzer.Test/Wrappers/RegisterSymbolStartActionWrapperTest.cs

```diff
@@ -300,38 +305,37 @@ End Sub
             End Class
             """, ignoreErrors: false, AnalyzerLanguage.VisualBasic);
         var visited = new List<string>();
-        var compilation = snippet.Compilation.WithAnalyzers(ImmutableArray.Create<DiagnosticAnalyzer>(
-            new TestDiagnosticAnalyzer(symbolStart =>
+        var compilation = snippet.Compilation.WithAnalyzers([new TestDiagnosticAnalyzer(symbolStart =>
             {
                 symbolStart.RegisterSyntaxNodeAction(syntaxNodeContext =>
                 {
                     var nodeName = syntaxNodeContext.Node.ToString();
                     visited.Add(nodeName);
                 }, VB.SyntaxKind.InvocationExpression);
-            }, SymbolKind.NamedType)));
+            }, SymbolKind.NamedType)]);
         var diagnostics = await compilation.GetAnalyzerDiagnosticsAsync();
-        visited.Should().BeEmpty();
         diagnostics.Should().BeEmpty();
+        visited.Should().BeEquivalentTo("ToString()");
     }
 
 #pragma warning disable RS1001 // Missing diagnostic analyzer attribute
 #pragma warning disable RS1025 // Configure generated code analysis
 #pragma warning disable RS1026 // Enable concurrent execution
-    private class TestDiagnosticAnalyzer : DiagnosticAnalyzer
+    private sealed class TestDiagnosticAnalyzer : DiagnosticAnalyzer
     {
+        public Action<SymbolStartAnalysisContextWrapper> Action { get; }
+        public SymbolKind SymbolKind { get; }
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics =>
+            [new DiagnosticDescriptor("TEST", "Test", "Test", "Test", DiagnosticSeverity.Warning, true)];
```

Comment:
You can (probably) use `AnalysisScaffolding.CreateDescriptor` to simplify this part. I didn't check all arguments and their implications.